components/proxy/plugins/configcat/configcat.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package configcat
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"go.uber.org/zap"
+	"golang.org/x/sync/singleflight"
+)
+
+const (
+	configCatModule = "gitpod.configcat"
+)
+
+var (
+	DefaultConfig = []byte("{}")
+	pathRegex     = regexp.MustCompile(`^/configcat/configuration-files/gitpod/config_v\d+\.json$`)
```

Comment:
In gitpod-desktop we use both keys production and non-production for testing, having `gitpod` harcoded only maps to one key at a time, maybe better to send the configcat key dinamically?

Reply:
I've thought about this before, we could 
- Create a new preview env to do this test. But it does make the steps more complicated
- Change `baseUrl` when creating configcat client and sdkKey if we want to connect to non-product env?

Reply:
yeah neither of those are nice :disappointed: 
- option 1 is definitely a no-no for me
- option 2, so we would use the configcat cdn as we do right now, correct?

cc @akosyakov thoughts?

Reply:
- Use product env in `vscode-desktop` is not harmful because configcat is calculating locally.
- If we bring non-product env to product gitpod, it will fetch non-product data every minute, but it's not need to do so
- Option 2 (change code if needed) can be a solution, i.e. use staging Gitpod as `baseUrl`

Reply:
@jeanp413  see https://github.com/gitpod-io/gitpod/pull/12756#issuecomment-1241671207